OrbitQt/AccessibilityAdapter.h

```diff
@@ -0,0 +1,188 @@
+#include "OrbitGlAccessibility.h"
+#include "orbitglwidget.h"
+
+namespace orbit_qt {
+
+/*Instances of this act as adapters from QAccessibleInterface to orbit_gl::GlAccessibleInterface and
+ vice versa. Static methods of A11yAdapter provide factory methods and tracking of adapters.
+
+ See file documentation above for more details.
+ */
+class A11yAdapter : public QAccessibleInterface {
+ public:
+  A11yAdapter() = delete;
+  A11yAdapter(A11yAdapter& rhs) = delete;
+  A11yAdapter(A11yAdapter&& rhs) = delete;
+
+  // vvv Overrides QAccessibleInterface vvv
+  bool isValid() const override {
+    bool result = info_ != nullptr;
+    CHECK(!result || interface_map_.find(info_)->second == this);
+    return result;
+  }
+  QObject* object() const override { return &dummy_; }
+  QAccessibleInterface* focusChild() const override { return nullptr; }
+
+  QAccessibleInterface* parent() const override {
+    return GetOrCreateAdapter(info_->AccessibleParent());
+  }
+  QAccessibleInterface* child(int index) const override {
+    return GetOrCreateAdapter(info_->AccessibleChild(index));
+  }
+  int childCount() const override { return info_->AccessibleChildCount(); }
+  int indexOfChild(const QAccessibleInterface* child) const override;
+  [[nodiscard]] QAccessibleInterface* childAt(int x, int y) const override;
+
+  QString text(QAccessible::Text /*t*/) const override { return info_->AccessibleName().c_str(); }
+  void setText(QAccessible::Text /*t*/, const QString& /*text*/) override{};
+  QRect rect() const override;
+  QAccessible::Role role() const override;
+
+  virtual QAccessible::State state() const override {
+    static_assert(sizeof(QAccessible::State) == sizeof(orbit_gl::A11yState));
+    auto state = info_->AccessibleState();
+    return *reinterpret_cast<QAccessible::State*>(&state);
+  }
+
+  // vvv Statics vvv
+  static QAccessibleInterface* GetOrCreateAdapter(const orbit_gl::GlAccessibleInterface* iface);
+  static void RegisterAdapter(const orbit_gl::GlAccessibleInterface* gl_control,
+                              QAccessibleInterface* qt_control) {
+    interface_map_.insert(std::make_pair(gl_control, qt_control));
+  }
+  // Called when a QAccessibleInterface which has been registered through "RegisterAdapter",
+  // but not created by this class, is deleted. Should only be needed for OrbitGlWidgets.
+  static void QAccessibleDeleted(QAccessibleInterface* iface);
+
+  static int RegisteredAdapterCount() { return interface_map_.size(); }
+
+ private:
+  mutable QObject dummy_;
+  A11yAdapter(const orbit_gl::GlAccessibleInterface* info) : info_(info){};
+
+  const orbit_gl::GlAccessibleInterface* info_ = nullptr;
+
+  static absl::flat_hash_map<const orbit_gl::GlAccessibleInterface*, QAccessibleInterface*>
+      interface_map_;
+  static bool initialized_;
+
+  static void Init();
+};
+
+/*Accessibility interface for OrbitGlWidget. See file documentation above for details.*/
+class OrbitGlWidgetAccessible : public QAccessibleWidget {
```

Comment:
Again, I'd argue it shouldn't, because both classes defined in this header are needed to achieve the core functionality of this, and are documented together in this header. The same actually goes for the contents of OrbitGlAccessibility.h, but in this case I needed to make the split because one is included in OrbitGl, and this one is not.

Reply:
The fact that two classes work together doesn't mean they should be in the same file. The only reason would be if `OrbitGlWidgetAccessible` was only used internally to `AccessibilityAdapter`. If I want to use `OrbitGlWidgetAccessible` I should include `OrbitGlWidgetAccessible.h`, not like the file named after the other class.

Reply:
As discussed offline, I've moved this (and the comments with it) into the .cpp file, because there is no need to expose it in the header to begin with.